### ISSUE TITLE
Accessibility improvements on Checkbox component

### ISSUE THREAD

**Issue by muratcorlu** [Trendyol/baklava#290]
[Figma](https://www.figma.com/file/RrcLH0mWpIUy4vwuTlDeKN/Baklava-Design-Guide?node-id=606%3A4458&t=EeXAS4ruHbwtJQAN-1)
- [ ] Checkbox component should be usable with keyboard navigation.
- [ ] Checkbox component should use related `aria-` attributes. [More info](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/checkbox_role)
